norse/task/cifar10.py

```diff
@@ -2,399 +2,212 @@
-            data, target = data.to(device), target.to(device)
-            output = model(data)
-            test_loss += torch.nn.functional.nll_loss(
-                output, target, reduction="sum"
-            ).item()  # sum up batch loss
-            pred = output.argmax(
-                dim=1, keepdim=True
-            )  # get the index of the max log-probabilioty
-            correct += pred.eq(target.view_as(pred)).sum().item()
-
-    test_loss /= len(test_loader.dataset)
-
-    accuracy = 100.0 * correct / len(test_loader.dataset)
-    logging.info(
-        f"\nTest set {FLAGS.model}: Average loss: {test_loss:.4f}, \
-            Accuracy: {correct}/{len(test_loader.dataset)} ({accuracy:.0f}%)\n"
-    )
-
-    if writer:
-        writer.add_scalar("Loss/test", test_loss, epoch)
-        writer.add_scalar("Accuracy/test", accuracy, epoch)
-
-    return test_loss, accuracy
-
+        y_hat, _ = torch.max(voltages, 0)
+        return y_hat
+
+    # Forward pass of a single batch
+    def training_step(self, batch, batch_idx):
+        x, y = batch
+        y_hat = self(x)
+        loss = label_smoothing_loss(y_hat, y)  # F.cross_entropy(y_hat, y)
+        acc1, acc5 = self.__accuracy(y_hat, y, topk=(1, 5))
+        self.log("train_loss", loss, on_step=True, on_epoch=True, logger=True)
+        self.log(
+            "train_acc1", acc1, on_step=True, prog_bar=True, on_epoch=True, logger=True
+        )
+        self.log("train_acc5", acc5, on_step=True, on_epoch=True, logger=True)
+        return loss
+
+    def validation_step(self, batch, batch_idx):
+        x, y = batch
+        y_hat = self(x)
+        loss = F.cross_entropy(y_hat, y)
+        acc1, acc5 = self.__accuracy(y_hat, y, topk=(1, 5))
+        self.log("val_loss", loss, on_step=True, on_epoch=True)
+        self.log("val_acc1", acc1, on_step=True, prog_bar=True, on_epoch=True)
+        self.log("val_acc5", acc5, on_step=True, on_epoch=True)
+
+    # The testing step is the same as the training, but with test data
+    def test_step(self, batch, batch_idx):
+        x, y = batch
+        y_hat = self(x)
+        loss = F.cross_entropy(y_hat, y)
+        acc1, acc5 = self.__accuracy(y_hat, y, topk=(1, 5))
+        self.log("test_loss", loss, on_step=True, on_epoch=True)
+        self.log("test_acc1", acc1, on_step=True, prog_bar=True, on_epoch=True)
+        self.log("test_acc5", acc5, on_step=True, on_epoch=True)
+
+    def training_epoch_end(self, outputs):
+        if self.lr_step:
+            self.scheduler.step()
+
+    def configure_optimizers(self):
+        if self.optimizer == "adam":
+            optimizer = torch.optim.Adam(
+                self.parameters(), lr=self.lr, weight_decay=1e-5
+            )
+        else:
+            optimizer = torch.optim.SGD(self.parameters(), lr=self.lr, momentum=0.9)
+        self.scheduler = torch.optim.lr_scheduler.StepLR(
+            optimizer, step_size=5, gamma=0.2
+        )
+        return optimizer
```

Comment:
You can let PyTorch-Lightning handle the Scheduler itself like such:
```python
return [optimizer], [scheduler]
```
And then delete the `training_epoch_end` method.